plugins/CoreHome/vue/src/ReportingMenu/ReportingMenu.store.ts

```diff
@@ -0,0 +1,231 @@
+        if (subcategory.isGroup) {
+          (subcategory.subcategories || []).forEach((subcat) => {
+            if (subcat.id === subcategoryId) {
+              foundCategory = category;
+              foundSubcategory = subcategory;
+              foundSubSubcategory = subcat;
+            }
+          });
+        }
+      });
+    });
+
+    return {
+      category: foundCategory,
+      subcategory: foundSubcategory,
+      subsubcategory: foundSubSubcategory,
+    };
+  }
+
+  private buildMenuFromPages() {
+    const menu = [];
+
+    const activeCategory = this.activeCategory.value;
+    const activeSubcategory = this.activeSubcategory.value;
+
+    const pages = ReportingPagesStoreInstance.pages.value;
+
+    const categoriesHandled = {};
+    pages.forEach((page) => {
+      const category = { ...page.category } as Category;
+      const categoryId = category.id;
+      const isCategoryActive = categoryId === activeCategory;
+
+      if (categoriesHandled[categoryId]) {
+        return;
+      }
+
+      categoriesHandled[categoryId] = true;
+
+      category.subcategories = [];
+
+      let categoryGroups: Subcategory;
+
+      const pagesWithCategory = pages.filter((p) => p.category.id === categoryId);
+      pagesWithCategory.forEach((p) => {
+        const subcategory = { ...p.subcategory } as Subcategory;
+        const isSubcategoryActive = subcategory.id === activeSubcategory && isCategoryActive;
+
+        if (p.widgets && p.widgets[0] && isNumeric(p.subcategory.id)) {
+          // we handle a goal or something like it
+          if (!categoryGroups) {
+            categoryGroups = { ...subcategory } as Subcategory;
+            categoryGroups.name = translate('CoreHome_ChooseX', [category.name]);
+            categoryGroups.isGroup = true;
+            categoryGroups.subcategories = [];
+            categoryGroups.order = 10;
+          }
+
+          if (isSubcategoryActive) {
+            categoryGroups.name = subcategory.name;
+          }
+
+          const entityId = page.subcategory.id;
+          subcategory.tooltip = `${subcategory.name} (id = ${entityId})`;
+
+          categoryGroups.subcategories.push(subcategory);
+          return;
+        }
+
+        category.subcategories.push(subcategory);
+      });
+
+      if (categoryGroups
+        && categoryGroups.subcategories
+        && categoryGroups.subcategories.length <= 5
+      ) {
+        categoryGroups.subcategories.forEach((sub) => category.subcategories.push(sub));
+      } else if (categoryGroups) {
+        category.subcategories.push(categoryGroups);
+      }
+
+      category.subcategories = this.sortMenuItems(category.subcategories);
+
+      menu.push(category);
+    });
+
+    return this.sortMenuItems(menu);
+  }
+
+  private sortMenuItems<T extends Orderable>(menu: T[]): T[] {
+    const result = [...menu];
+    result.sort((lhs, rhs) => {
+      if (lhs.order < rhs.order) {
+        return -1;
+      }
+
+      if (rhs.order > lhs.order) {
```

Comment:
Isn't that the same as the check above? Guess it should be
```suggestion
      if (lhs.order > rhs.order) {
```

Reply:
Good catch, thanks :+1: